### ISSUE TITLE
Does Flutter WebRTC support H.265 ?

### ISSUE THREAD

**Issue by xieenming** [flutter-webrtc/flutter-webrtc#862]
Hello, I would like to know if flutter-webrtc supports [H.265](https://en.wikipedia.org/wiki/High_Efficiency_Video_Coding) ?

Thanks a lot!

**Comment by cloudwebrtc** [comment]
Not yet supported, but it can be supported by patching the [webrtc-sdk](https://github.com/webrtc-sdk/webrtc) source code.

**Comment by xieenming** [comment]
> Not yet supported, but it can be supported by patching the [webrtc-sdk](https://github.com/webrtc-sdk/webrtc) source code.
> 
> Currently, flutter-webrtc uses the webrtc source code maintained by the community.

Thanks a lot for your quick reply.

**Comment by boletoms** [comment]
@cloudwebrtc any update on this?
do we still need to patch it?

**Comment by KlGleb** [comment]
Hello, can I ask you if it is supported or not at the moment?

**Comment by xuwucv** [comment]
Hello, can I ask you if it is supported or not at the moment?

**Comment by cloudwebrtc** [comment]
we plan to upgrade webrtc to m135+ in 1-2 months, which will support H265 by default.

**Comment by xuwucv** [comment]
@cloudwebrtc Thank you for your reply. Looking forward to your new version

**Comment by qminhminh** [comment]
HI

**Comment by qminhminh** [comment]
HI, currently I connect to the camera as H.264, is there any way to switch to H.265, I code in Flutter language (flutter_webrtc)

**Comment by kyjcoope** [comment]
@cloudwebrtc Any word on H.265 support and the upgrade to m135+?

**Comment by cloudwebrtc** [comment]
Yeah, we are upgrading to m137, once completed we will be able to support H.265
https://github.com/webrtc-sdk/webrtc/pull/173

**Comment by ozdonner1** [comment]
@cloudwebrtc is there any time estimation when the upgrade to m137 will be completed?

**Comment by iotmehedi** [comment]
I am using Flutter_webrtc version flutter_webrtc: ^0.14.2. but still not support H265. Is there have any procedure to support or by default not added in update version?

**Comment by CaptainTeemo** [comment]
> I am using Flutter_webrtc version flutter_webrtc: ^0.14.2. but still not support H265. Is there have any procedure to support or by default not added in update version?

@iotmehedi 
I forked the [webrtc](https://github.com/CaptainTeemo/webrtc) repo and made it support h265 on iOS https://github.com/CaptainTeemo/flutter-webrtc for fun, you can add 
```
flutter_webrtc:
    git:
      url: https://github.com/CaptainTeemo/flutter-webrtc.git
      ref: 'main'
```
to the `pubspec.yaml` to have a try.

**Comment by iotmehedi** [comment]
> > I am using Flutter_webrtc version flutter_webrtc: ^0.14.2. but still not support H265. Is there have any procedure to support or by default not added in update version?
> 
> [@iotmehedi](https://github.com/iotmehedi) I forked the [webrtc](https://github.com/CaptainTeemo/webrtc) repo and made it support h265 on iOS https://github.com/CaptainTeemo/flutter-webrtc for fun, you can add
> 
> ```
> flutter_webrtc:
>     git:
>       url: https://github.com/CaptainTeemo/flutter-webrtc.git
>       ref: 'main'
> ```
> 
> to the `pubspec.yaml` to have a try.

Thanks a lot brother.
 but have any changes or challenges for android? Actually I need for both platform.

**Comment by misakayaho** [comment]
Hello, can I ask you if it is supported or not at the moment?
I am using Flutter_webrtc version flutter_webrtc: ^1.0.0. but still not support H265. Is there have any procedure to support or by default not added in update version?

**Comment by cloudwebrtc** [comment]
we are adding H.265 support to libwebrtc and fixing related features (E2EE), which should be applied to flutter-webrtc in a week or two

**Comment by misakayaho** [comment]
> we are adding H.265 support to libwebrtc and fixing related features (E2EE), which should be applied to flutter-webrtc in a week or two

Thank you for your reply. Looking forward to your new version

**Comment by qminhminh** [comment]
hi when you have update about new flutter webrtc library then please let us
know

**Comment by qminhminh** [comment]
@misakayaho @KlGleb @CaptainTeemo @boletoms @kyjcoope 
Hello everyone,

I am developing a Flutter mobile application that connects to an IP surveillance camera which already supports two-way audio (microphone & speaker).
Currently, I can receive video and audio from the camera in my app via WebRTC .

What I need now is Push-to-Talk functionality:

When the user presses and holds a button, the app should capture audio from the phone's microphone and send it to the camera's speaker in real time.

When the user releases the button, audio transmission should stop.

My environment:

Flutter version: 3.27.4

**Comment by qminhminh** [comment]
> we are adding H.265 support to libwebrtc and fixing related features (E2EE), which should be applied to flutter-webrtc in a week or two

Users/minhhaquang/.pub-cache/hosted/pub.dev/flutter_webrtc-1.1.0/android/src/main/java/com/cloudwebrtc/webrtc/SurfaceTextureRenderer.java:118: error: <anonymous com.cloudwebrtc.webrtc.SurfaceTextureRenderer$1> is not abstract and does not override abstract method onSurfaceDestroyed() in Callback
            new TextureRegistry.SurfaceProducer.Callback() {
                                                           ^
/Users/minhhaquang/.pub-cache/hosted/pub.dev/flutter_webrtc-1.1.0/android/src/main/java/com/cloudwebrtc/webrtc/SurfaceTextureRenderer.java:124: error: method does not override or implement a method from a supertype
              @Override
              ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors
3 warnings

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':flutter_webrtc:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --info option to get more log output.
> Run with --scan to get full insights.

BUILD FAILED in 33s
Error: Gradle task assembleDebug failed with exit code 1

**Comment by Hackmodford** [comment]
> we are adding H.265 support to libwebrtc and fixing related features (E2EE), which should be applied to flutter-webrtc in a week or two

Was there any progress made in this effort?
